src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,852 @@
+        }
+
+        for( FeatureInput<VariantContext> variantFeature : resource ) {
+            dataManager.addTrainingSet( new TrainingSet( variantFeature ) );
+        }
+
+        if( !dataManager.checkHasTrainingSet() ) {
+            throw new UserException.CommandLineException(
+                    "No training set found! Please provide sets of known polymorphic loci marked with the training=true ROD binding tag. For example, -resource:hapmap,VCF,known=false,training=true,truth=true,prior=12.0 hapmapFile.vcf" );
+        }
+        if( !dataManager.checkHasTruthSet() ) {
+            throw new UserException.CommandLineException(
+                    "No truth set found! Please provide sets of known polymorphic loci marked with the truth=true ROD binding tag. For example, -resource:hapmap,VCF,known=false,training=true,truth=true,prior=12.0 hapmapFile.vcf" );
+        }
+
+        //TODO: this should be refactored/consolidated as part of
+        // https://github.com/broadinstitute/gatk/issues/121 and
+        // https://github.com/broadinstitute/gatk/issues/1116
+        // Initialize VCF header lines
+        final Set<VCFHeaderLine> hInfo = new HashSet<>();
+        hInfo.add(new VCFHeaderLine("source", this.getClass().getSimpleName()));
+        VariantRecalibrationUtils.addVQSRStandardHeaderLines(hInfo);
+        Set<VCFHeaderLine> actualLines = null;
+        SAMSequenceDictionary sequenceDictionary = null;
+        if (hasReference()) {
+            File refFile = referenceArguments.getReferenceFile();
+            sequenceDictionary= this.getReferenceDictionary();
+            actualLines = VcfUtils.updateHeaderContigLines(hInfo, refFile, sequenceDictionary, true);
+        }
+        else {
+            sequenceDictionary = getHeaderForVariants().getSequenceDictionary();
+            if (null != sequenceDictionary) {
+                actualLines = VcfUtils.updateHeaderContigLines(hInfo, null, sequenceDictionary, true);
+            }
+            else {
+                actualLines = hInfo;
+            }
+        }
+        //TODO: integrate lenient argument
+        recalWriter = GATKVariantContextUtils.createVCFWriter(new File(output), getBestAvailableSequenceDictionary(), true);
+        recalWriter.writeHeader( new VCFHeader(actualLines) );
+
+        for( int iii = 0; iii < REPLICATE * 2; iii++ ) {
+            replicate.add(Utils.getRandomGenerator().nextDouble());
+        }
+    }
+
+    //---------------------------------------------------------------------------------------------------------------
+    //
+    // map
+    //
+    //---------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public void apply(VariantContext vc, ReadsContext readsContext, ReferenceContext ref, FeatureContext featureContext) {
+
+        // TODO: this variant came from the drivingvariants input; do we still need to query it again for overlapping ?
```

Comment:
Changed this to accumulate and queue variantContexts with the same start-location, and then use the saved vc/featureContext pairs directly, followed by a single call to query the overlapping aggregates once for the group, which I believe matches the behavior of GATK3.